app/fruits/[fruitId]/page.js

```diff
@@ -0,0 +1,66 @@
+import { notFound } from 'next/navigation';
+import { getFruitById } from '../../../database/fruits';
+import { getCookie } from '../../../util/cookies';
+import { parseJson } from '../../../util/json';
+import { rootNotFoundMetadata } from '../../not-found';
+import FruitCommentForm from './FruitCommentForm';
+
+// export const fruits = [
+//   { id: 1, name: 'Banana', icon: '🍌' },
+//   { id: 2, name: 'Coconuts', icon: '🥥' },
+//   { id: 3, name: 'Papaya', icon: '🥔' },
+//   { id: 4, name: 'Mango', icon: '🥭' },
+//   { id: 5, name: 'Avocado', icon: '🥑' },
+// ];
+
+export function generateMetadata({ params }) {
+  const fruit = getFruitById(Number(params.fruitId));
+
+  if (!fruit) {
+    return rootNotFoundMetadata;
+  }
+
+  return {
+    title: fruit.name,
+    description: `Single fruit page for ${fruit.name}`,
+  };
+}
+
+// we add this only if we have no dynamic function as cookies or headers
+export const dynamic = 'force-dynamic';
```

Comment:
do we need this? isn't this handled already by the call to `cookies()` in the imported `util/cookies` file? (because it's being imported in this file)

Reply:
we decided to leave it out of the lecture until it's needed